src/chunked_scatter/chunked_scatter.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2019 Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+from pathlib import Path
+from typing import TextIO
+
+
+def dict_chunker(in_file: TextIO, chunk_size: int, overlap: int):
+    """
+    Chunk the chromosomes/contigs from a dict.
+    :param in_file: The input file.
+    :param chunk_size: The size of the chunks.
+    :param overlap: The size of the overlap between chunks.
+    """
+    for line in in_file.readlines():
+        line = line.split()
+        if line[0] == "@SQ":
+            for field in line:
+                if field[:2] == "LN":
+                    length = int(field.split(":")[1])
+                elif field[:2] == "SN":
+                    name = ":".join(field.split(":")[1:])
+            # This will cause the last chunk to be between 0.5 and 1.5
+            # times the chunk_size in length, this way we avoid the
+            # possibility that the last chunk ends up being to small
+            # (eg. 1+overlap bases).
+            position = 0
+            while position + chunk_size*1.5 < length:
+                if position-overlap <= 0:
+                    yield [name, 0, int(position+chunk_size)]
+                else:
+                    yield [name, int(position-overlap), int(position+chunk_size)]
+                position += chunk_size
+            if position-overlap <= 0:
+                yield [name, 0, length]
+            else:
+                yield [name, int(position-overlap), length]
+
+
+def bed_chunker(in_file: TextIO, chunk_size: int, overlap: int):
+    """
+    Chunk the entries of the bed file.
+    :param in_file: The input file.
+    :param chunk_size: The size of the chunks.
+    :param overlap: The size of the overlap between chunks.
+    """
+    for line in in_file.readlines():
+        line = line.strip().split("\t")
```

Comment:
Same comment as above. Please do not replace line with line. Better to call the first `line` `linestring` or `raw_line` or something.